driver_test.go

```diff
@@ -2189,6 +2189,56 @@ func TestEmptyPassword(t *testing.T) {
 	}
 }
 
+func TestConnectAttrs(t *testing.T) {
+	if !available {
+		t.Skipf("MySQL server not running on %s", netAddr)
+	}
+
+	db, err := sql.Open("mysql", dsn+"&connectAttrs=program_name:GoTest,foo:bar")
+	if err != nil {
+		t.Fatalf("error connecting: %s", err.Error())
+	}
+	defer db.Close()
+	dbt := &DBTest{t, db}
+
+	rows := dbt.mustQuery("SHOW VARIABLES LIKE 'performance_schema'")
+	if rows.Next() {
+		var var_name, value string
+		rows.Scan(&var_name, &value)
+		if value != "ON" {
+			t.Skip("performance_schema is disabled")
+		}
+	} else {
+		t.Skip("no performance_schema variable in mysql")
+	}
+
+	rows, err = dbt.db.Query("SELECT attr_value FROM performance_schema.session_connect_attrs WHERE processlist_id=CONNECTION_ID() AND attr_name='program_name'")
+	if err != nil {
+		dbt.Skipf("server probably does not support performance_schema.session_connect_attrs: %s", err)
+	}
+
+	if rows.Next() {
+		var str string
+		rows.Scan(&str)
+		if "GoTest" != str {
+			dbt.Errorf("GoTest != %s", str)
+		}
+	} else {
+		dbt.Error("no data for program_name")
+	}
+
```

Comment:
Put `rows.Close()` here.

Reply:
Done in https://github.com/go-sql-driver/mysql/pull/1241/commits/406e2a1c5330545ce9c2a7df6c89cd31dde85c2d